dxr/vcs.py

```diff
@@ -0,0 +1,326 @@
+"""Let DXR understand the concept of version control systems. The main entry points are `tree_to_repos`, which produces a mapping of roots to VCS objects
+for each version control root discovered under the provided tree, and
+`path_to_vcs`, which returns a VCS object for the version control system that
+tracks the given path. Currently supported VCS are Mercurial, Git, and
+Perforce.
+"""
+import marshal
+import os
+from os.path import relpath, join, dirname
+import subprocess
+import urlparse
+
+import hglib
+from ordereddict import OrderedDict
+from warnings import warn
+
+import dxr
+
```

Comment:
And then, for consistency (foremost importance) and PEP 8 (secondary), we skip 2 lines after the imports and between every top-lvel construct (like classes or module-level functions).